packages/core/src/extensions/Placeholder/PlaceholderPlugin.ts

```diff
@@ -16,7 +16,12 @@ export const PlaceholderPlugin = (
       if (nonce) {
         styleEl.setAttribute("nonce", nonce);
       }
-      document.head.appendChild(styleEl);
+      if (editor._tiptapEditor.view.root instanceof ShadowRoot) {
+        editor._tiptapEditor.view.root.append(styleEl);
+      } else if (editor._tiptapEditor.view.root instanceof Document) {
```

Comment:
Don't need `else if` here, `else` should be fine given the typing. Same goes in a few other places where you check `instanceof`.